src/native/fs/fs_napi.cpp

```diff
@@ -473,7 +487,7 @@ struct Readfile : public FSWorker
     }
     virtual void Work()
     {
-        int fd = open(_path.c_str(), O_RDONLY);
+        int fd = open(_path.c_str(), O_RDONLY, _mode);
```

Comment:
Maybe I'm missing something here, but why do you need to pass a mode to open on read? Isn't that only relevant when you use O_CREAT?